test/jdk/java/util/Locale/CaseFoldLanguageTagTest.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8159337
+ * @summary Test Locale.caseFoldLanguageTag(String languageTag)
+ * @run junit caseFoldLanguageTagTest
+ */
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.Locale;
+import java.util.stream.Stream;
+
+/**
+ * Test the implementation of Locale.caseFoldLanguageTag(String languageTag).
+ * A variety of well-formed tags are tested, composed of the following subtags:
+ * language, extlang, script, region, variant, extension, singleton, privateuse,
+ * grandfathered, and irregular. For more info, see the following,
+ * <a href="https://www.rfc-editor.org/rfc/rfc5646.html#section-2.1">Tag Syntax</a>).
+ * In addition, the method is tested to ensure that IllegalArgumentException and
+ * NullPointerException are thrown given the right circumstances.
+ */
+public class caseFoldLanguageTagTest {
+
+    @ParameterizedTest
+    @MethodSource("wellFormedTags")
+    public void TestWellFormedTags(String tag, String foldedTag) {
+        assertEquals(foldedTag, Locale.caseFoldLanguageTag(tag), String.format("Folded %s", tag));
+    }
+
+    @ParameterizedTest
+    @MethodSource("illFormedTags")
+    public void TestIllFormedTags(String tag) {
+        try {
+            Locale.caseFoldLanguageTag(tag);
+            throw new RuntimeException("$$$ Failure: ill formed tags should throw IAE");
+        } catch (IllegalArgumentException ignored) {}
```

Comment:
`IllformedLocaleException`? Or if the test passes with this version of impl, should the test throw an RuntimeException? Also instead of try-catch, `assertThrows()` can be used.

Reply:
The test does not pass like this, I had forgotten to re run the test after originally switching IAE to IFE in the method itself. (and hence why I did not catch that it is not updated in the test). 

I have switched all the IAE to IFE in the test, and also used assertThrows() as recommended